docs/05-infrastructure-nodes.md

```diff
@@ -0,0 +1,318 @@
+
+In our case, we want to ensure that the resulting node inherits one or more specific labels. As you've seen in the examples above, labels go in `metadata` sections:
+
+```YAML
+  spec:
+      metadata:
+        creationTimestamp: null
+      providerSpec:
+        value:
+          ami:
+            id: ami-08871aee06d13e584
+....
+```
+
+By default the `MachineSets` that the installer creates do not apply any
+additional labels to the node.
+
+## Defining a Custom MachineSet
+Now that you've analyzed an existing `MachineSet` it's time to go over the
+rules for creating one, at least for a simple change like we're making:
+
+1. Don't change anything in the `providerSpec`
+1. Don't change any instances of `sigs.k8s.io/cluster-api-cluster: <clusterid>`
+1. Give your `MachineSet` a unique `name`
+1. Make sure any instances of `sigs.k8s.io/cluster-api-machineset` match the `name`
+1. Add labels you want on the nodes to `.spec.template.spec.metadata.labels`
+1. Even though you're changing `MachineSet` `name` references, be sure not to change the `subnet`.
+
+This sounds complicated, so let's go through an example. Go ahead and dump
+one of your existing `MachineSets` to a file, and then open it with your
+favorite text editor. For example:
+
+    oc get machineset 190125-3-worker-us-west-1b -o yaml -n openshift-cluster-api > infra-machineset.yaml
+
+### Clean It
+Since we asked OpenShift to tell us about an existing `MachineSet`, there's a
+lot of extra data that we can immediately remove from the file. At the
+`.metadata` top level, delete:
+
+1. `creationTimestamp`
+1. `generation`
+1. `resourceVersion`
+1. `selfLink`
+1. `uid`
+
+Then, delete the entire `.status` block.
+
+### Name It
+Go ahead and change the top-level `.metadata.name` to something indicative of
+the purpose of this set, for example:
+
+    name: infrastructure-us-west-1b
+
+By looking at this `MachineSet` we can tell that it houses
+infrastructure-focused `Machines` (nodes) in `us-west-1` in the `b`
+availability zone. Of course, you will want to change this to something that
+makes sense for your cluster.
+
+### Match It
+Change any instance of `sigs.k8s.io/cluster-api-machineset` to match your new
+name of `infrastructure-us-west-1b`. This appears in both
+`.spec.selector.matchLabels` as well as `.spec.template.metadata.labels`.
+
+### Add Your Node Label
+Since we are adding nodes for infrastructure, a simple label that might make sense is `infra=true`. One caveat to this is that `true` is considered a boolean and not a string.
+
+Add a `labels` section to `.spec.template.spec.metadata` with the label `infra: "true"` (the quotes are because of the boolean). It should look like the following:
+
+```YAML
+    spec:
+      metadata:
+        creationTimestamp: null
```

Comment:
nit: you can drop `creationTimestamp` (and maybe mention it when you talk about removing the top-level `creationTimestamp` above?).

Reply:
i thought back and forth on that. I decided not to explicitly remove the null ones because i didn't want to give people extra work to do. but I guess I could also say "remove all instances of creationtimestamp" too... 

I'll fix that.